aws_route53_record.test_cert_validation issue with aws provider

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version
Terraform v0.11.14
+ provider.aws v2.21.1
+ provider.null v1.0.0



<!---

 --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* aws_route53_record, aws_acm_certificate

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

terraform apply output below....

data.aws_route53_zone.cluster_zone: Refreshing state...
aws_acm_certificate.test_cert: Creating...
  arn:                         "" => "<computed>"
  domain_name:                 "" => "<redacted>"
  domain_validation_options.#: "" => "<computed>"
  subject_alternative_names.#: "" => "<computed>"
  tags.%:                      "" => "5"
  .
  .
  .
  validation_emails.#:         "" => "<computed>"
  validation_method:           "" => "DNS"
aws_acm_certificate.test_cert: Creation complete after 3s (ID: <redacted>)

Error: Error applying plan:

1 error occurred:
* aws_route53_record.test_cert_validation: Resource 'aws_acm_certificate.test_cert' does not have attribute 'domain_validation_options.0.resource_record_value' for variable 'aws_acm_certificate.test_cert.domain_validation_options.0.resource_record_value'

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

<!--- What should have happened? --->

should create the aws_route53_record


### Actual Behavior

<!--- What actually happened? --->

1 error occurred:
* aws_route53_record.test_cert_validation: Resource 'aws_acm_certificate.test_cert' does not have attribute 'domain_validation_options.0.resource_record_value' for variable 'aws_acm_certificate.test_cert.domain_validation_options.0.resource_record_value'

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

on the following code replacing redacted values with suitable values.

provider "aws" {
  region = "ap-southeast-2"
  version = "~> 2.21.1"
  assume_role {
    role_arn = "arn:aws:iam::<redacted>:role/<redacted>"
    session_name = "SESSION_NAME"
    external_id = "EXTERNAL_ID"
  }
}

resource "aws_acm_certificate" "test_cert" {
  domain_name = " <redacted>"
  validation_method = "DNS"
  tags {
    <redacted>
  }
}

data "aws_route53_zone" "cluster_zone" {
  name = "<redacted>"
  private_zone = false
}

resource "aws_route53_record" "test_cert_validation" {
 name = "${aws_acm_certificate.test_cert.domain_validation_options.0.resource_record_name}"
 type = "${aws_acm_certificate.test_cert.domain_validation_options.0.resource_record_type}"
 zone_id = "${data.aws_route53_zone.cluster_zone.zone_id}"
 records = ["${aws_acm_certificate.test_cert.domain_validation_options.0.resource_record_value}"]
 ttl = 60
}

resource "aws_acm_certificate_validation" "test_cert" {
  certificate_arn = "${aws_acm_certificate.test_cert.arn}"
  validation_record_fqdns = ["${aws_route53_record.test_cert_validation.fqdn}"]
}

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->


* #0000

This appears to be a region specific problem. If I run the acceptance test `TestAccAWSAcmCertificate_dnsValidation` in the default region (`us-west-2`) it works fine. If I then set the region to `ap-southeast-2` by exporting `AWS_DEFAULT_REGION`, I get the following error:

```
--- FAIL: TestAccAWSAcmCertificate_dnsValidation (19.56s)
    testing.go:568: Step 0 error: Check failed: Check 2/10 error: aws_acm_certificate.cert: Attribute 'domain_name' expected "tf-acc-6668933416344597644.<REDACTED>", got ""
FAIL
```

--- a/aws/resource_aws_acm_certificate.go
+++ b/aws/resource_aws_acm_certificate.go
@@ -292,6 +292,10 @@ func convertValidationOptions(certificate *acm.CertificateDetail) ([]map[string]
 	var emailValidationResult []string
 
 	if *certificate.Type == acm.CertificateTypeAmazonIssued {
+		if len(certificate.DomainValidationOptions) == 0 {
+			log.Printf("[DEBUG] No validation options need to retry.")
+			return nil, nil, fmt.Errorf("No validation options need to retry.")
+		}
 		for _, o := range certificate.DomainValidationOptions {
 			if o.ResourceRecord != nil {
 				validationOption := map[string]interface{}{

